source/data-modeling/associations.txt

```diff
@@ -0,0 +1,504 @@
+this guide, you can learn about the different types of associations that
+{+odm+} supports and how to use them in your application.
+
+Referenced Associations
+------------------------
+
+Referenced associations allow you to create a relationship between two models
+where one model references the other. {+odm+} supports the following referenced
+association types:
+
+- ``has_one``
+- ``has_many``
+- ``belongs_to``
+- ``has_and_belongs_to_many``
+
+The following sections describe how to use each of these association types.
+
+Has One
+~~~~~~~
+
+You can use the ``has_one`` macro to declare that a class model contains a child document
+that is stored in a separate collection. The following example creates a
+``Band`` class with a ``has_one`` relationship to a ``Studio`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-one
+   :end-before: # end-has-one
+
+When you declare a ``has_one`` association, the child class must also use the
+``belongs_to`` association that references the parent class. The following
+example shows the ``Studio`` class referenced in the preceding ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-one-child
+   :end-before: # end-has-one-child
+
+To learn more about the ``belongs_to`` macro, see the :ref:`Belongs To
+<mongoid-belongs-to>` section.
+
+You can use :ref:`validations <mongoid-validation>` to ensure that the child
+class is present in your parent class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-one-validation
+   :end-before: # end-has-one-validation
+   :emphasize-lines: 6
+
+Has Many
+~~~~~~~~
+
+You can use the ``has_many`` macro to declare that a class model contains
+multiple child documents that are stored in a separate collection. The following
+example creates a ``Band`` class with a ``has_many`` relationship to a
+``Members`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-many
+   :end-before: # end-has-many
+
+When you declare a ``has_many`` association, the child class must also use the
+``belongs_to`` association that references the parent class. The following
+example shows the ``Member`` class referenced in the preceding ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-many-child
+   :end-before: # end-has-many-child
+
+To learn more about the ``belongs_to`` macro, see the :ref:`Belongs To
+<mongoid-belongs-to>` section.
+
+You can use :ref:`validations <mongoid-validation>` to ensure that the child
+class is present in your parent class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-many-validation
+   :end-before: # end-has-many-validation
+   :emphasize-lines: 6
+
```

Comment:
S: since there's a lot of info below about any? and exists?, you could make a subsection here titled something like "Association methods" or "Retrieve association information"